src/Aspire.Hosting/ApplicationModel/AIOptOutAnnotation.cs

```diff
@@ -0,0 +1,8 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+internal sealed class AIOptOutAnnotation : IResourceAnnotation
```

Comment:
My preference would go to using a name with `MCP` instead of `AI`, and make it more generic so it can be use for more than just opt-out. `McpSettingsAnnotation` (settings, configuration, options, preferences, ...).

This way it can have a `Disable` property, but we could also decide to have `DisableLogs`, ... Or use flags with all the things we expose overtime.

This also allows us to postpone the granularity and just have an all-out setting for now. Making an `All` enum doesn't prevent us from adding new specific values later.